### PR TITLE
feat/(AUTH-CPC-1352): Find User From all users and be able to see and modify it (Exluding password and roles)

### PR DESCRIPTION
**Pull request by Justin222993**
**JIRA:** [link to jira ticket](https://champlainsaintlambert.atlassian.net/jira/software/c/projects/CPC/boards/15/backlog?epics=visible&selectedIssue=CPC-1352&text=Auth)
## Context:
As an admin, I want to be able to get a user’s information by id and and be able to modify the data within it,
this ticket will change the getAllUsers page, add new edit and user pages and the entire back-end for updating a user's information
## Does this PR change the .vscode folder in petclinic-frontend?:
It does not

<span style="color:red">**Reviewers need to check for any changes to the 
.vscode folder and add a comment about it to their review comments.**</span>

## Changes
- Added update user function in api gateway (Didn't exist)
- Added update user endpoint in api gateway (Didn't exist)
- Added update user function in auth service (Didn't exist)
- Added update user endpoint in auth service (Didn't exist)
- Added UpdateUserForm.tsx + css
- Added UserDetails.tsx + css
- Modified AllUsers css to better fit the other css
- Added 2 tests in api gateway v2/UserControllerUnitTest.java
- Added 4 tests in auth service UserControllerIntegrationTest.java
- Added links to find the right pages
## Before and After UI (Required for UI-impacting PRs)
Before get all users page: 
image

New get all users page:
image

New user details page:
image

User details page AS OWNER:
image

New User Form page:
image

Form Validation:
image

Form Modal:
image

Updated info:
image

## Dev notes
Roles will be added in a future pull request

### CODE REVIEW

File: auth-service/src/main/java/com/petclinic/authservice/presentationlayer/User/UserController.java [hunk 1]
```diff
@@ -104,6 +104,10 @@ public ResponseEntity<UserPasswordLessDTO> updateUserRole (@PathVariable String
         return ResponseEntity.ok().body(userService.updateUserRole(userId, roleChanged, token));
     }
 
+    @PutMapping("/{userId}")
+    public ResponseEntity<User> updateUser (@PathVariable String userId, @RequestBody UserPasswordLessDTO userPasswordLessDTO) {
+        return ResponseEntity.ok().body(userService.updateUser(userId, userPasswordLessDTO));
+    }
```
Comment (NotZagreus) [inline auth-service/src/main/java/com/petclinic/authservice/presentationlayer/User/UserController.java:110]:
```suggestion
   public Mono<UserPasswordLessDTO> updateUser(@PathVariable String userId, @RequestBody @Valid UserPasswordLessDTO model, @CookieValue("Bearer") String jwtToken) {
    return authServiceClient.updateUser(userId, model, jwtToken);
}
```
You can add validation for the UserPasswordLessDTO model in the updateUser method to ensure data integrity.

**Review by NotZagreus — APPROVED** [review]
My only suggestion was to add the UserPasswordLessDTO model in the updateUser method to ensure data integrity, except for that it looks very nice.

File: petclinic-frontend/src/router.tsx
Comment (FelixAllard) [inline petclinic-frontend/src/router.tsx:1]: You did it the right way ! There should not be any problem in the NavBar After this

**Review by FelixAllard — APPROVED** [review]
Looks like everything is in place! Good job!

File: petclinic-frontend/src/router.tsx
Comment (Justin222993) [reply petclinic-frontend/src/router.tsx:1]: lol

File: auth-service/src/test/java/com/petclinic/authservice/presentationlayer/User/UserControllerIntegrationTest.java
[the remark this answers is not in the document]
Comment (Justin222993) [reply auth-service/src/test/java/com/petclinic/authservice/presentationlayer/User/UserControllerIntegrationTest.java:1]: Thanks

**Pull request merged**